modules/openapi-generator/src/main/resources/python/model_templates/method_from_openapi_data_composed.mustache

```diff
@@ -48,7 +48,7 @@
             '_visited_composed_classes': self._visited_composed_classes,
         }
         composed_info = validate_get_composed_info(
-            constant_args, kwargs, self)
+            constant_args, kwargs, self, from_openapi_data=True)
```

Comment:
Why are we passing in from_openapi_data=True here?
Isn't the _from_openapi_data method only used when receiving responses from the server.
When that is done _from_openapi_data is invoked which means __init__ is NOT invoked.
So this code path is for client side instantiation where from_openapi_data is False, right?

Reply:
That is a good question. I suspect the orignal intention was that the method from_openapi_data is data from server. However, it can also be used to instantiate objects where you might mocking a server (ie. generating test code, generating an object with some data that needs its read-only data set). In this particular case aboe that you raised, this is a copy. Since we know the object was rendered already (from whereever, client or server), we just need to copy it. If we used the normal __init__ method, if there is read-only data, it will raise an exception. Rather, this this case, we just need it copied, as is, without any checking. This is why the from_openapi_data is choosen.

Reply:
Since this is the generated code for from_openapi_data, we need to let the method validate_get_composed_info know this, since it eventually calls get_allof_instances, what then may need to instantiate allOf objects.  While instantiating the allOf objects, it needs to instantiate the using the appropriate init method (__init__, or __from_openapi_data).

With regard to your comment on from client or server (and using from_openapi_data), _from_openapi_data can also be used to instantiate objects where you might mocking a server (ie. generating test code, generating an object with some data that needs its read-only data set).